File: school-management-backend/src/main/java/com/schoolbus/service/AuthService.java
package com.schoolbus.service;

import com.schoolbus.dto.request.LoginRequest;
import com.schoolbus.dto.response.AuthResponse;
import com.schoolbus.dto.response.UserResponse;

public interface AuthService {
    AuthResponse login(LoginRequest request, String ipAddress, String deviceInfo);
    AuthResponse refresh(String refreshToken, String ipAddress, String deviceInfo);
    void logout(String refreshToken);
    UserResponse getCurrentUser();
}

File: school-management-frontend/src/services/auth.service.ts
import { api } from "./api";
import type { ApiResponse } from "@/types/api";
import type { AuthResponse, LoginPayload } from "@/types/auth";
import type { User } from "@/types/user";

export const authService = {
  login: async (payload: LoginPayload) => (await api.post<ApiResponse<AuthResponse>>("/api/v1/auth/login", payload)).data.data,
  refresh: async (refreshToken: string) => (await api.post<ApiResponse<AuthResponse>>("/api/v1/auth/refresh", { refreshToken })).data.data,
  logout: async (refreshToken: string) => api.post("/api/v1/auth/logout", { refreshToken }),
  me: async () => (await api.get<ApiResponse<User>>("/api/v1/auth/me")).data.data,
};
